libraries/ms-common/inc/spi.h

```diff
@@ -0,0 +1,111 @@
+// SPI HAL Interface
+
+#include "interrupt.h"
+#include "status.h"
+#include "stm32f0xx.h"
+
+typdef void (*spi_callback)(SPIPeriph spi_x, void *context);
+
+// For selecting which peripheral
+typedef enum {
+  SPI_PERIPH_1 = 0,
+  SPI_PERIPH_2,
+} SPIPeriph;
```

Comment:
I would personally replace this with a `uint8_t`. Most architectures will have a variable number of SPI peripherals so constraining this to an enum makes for weird cases where you have to return `STATUS_CODE_INVALID_ARGS` for an otherwise valid input because the microcontroller doesn't support that number of peripherals.